base/test.jl

```diff
@@ -271,6 +276,17 @@ if an exception is thrown inside the test block but outside of a `@test` context
 function record end
 
 """
+    should_run(ts::AbstractTestSet, nruns::Int)
+
+Decide if the tests should be run; returns true if and only if they should.
+The tests are called in a loop and the `nruns` arg counts the number of times the test
+block has been executed. `nruns` is 0 when the tests have not yet executed.
+"""
+function should_run(ts::AbstractTestSet, nruns::Int)
```

Comment:
Seems kind of weird to define only this method for `AbstractTestSet` - is the intention that most `TestSets` will not implement this method themselves?